src/py/reactpy/reactpy/core/vdom.py

```diff
@@ -328,7 +328,7 @@ def _validate_child_key_integrity(value: Any) -> None:
         )
     else:
         for child in value:
-            if isinstance(child, ComponentType) and child.key is None:
+            if child.key is None and isinstance(child, ComponentType):
```

Comment:
Issue L - perf: since this is a warning condition, the cheaper check is done first.